smoketest/graphqlnotifiedalerts_test.go

```diff
@@ -0,0 +1,171 @@
+	values ({{uuid "esid"}}, {{uuid "eid"}});
+ 
+	insert into escalation_policy_actions (id, escalation_policy_step_id, user_id)
+	values ({{uuid "epa"}}, {{uuid "esid"}}, {{uuid "user"}});
+
+	insert into services (id, escalation_policy_id, name) 
+	values
+		({{uuid "sid"}}, {{uuid "eid"}}, 'service'),
+		({{uuid "sid2"}}, {{uuid "eid2"}}, 'service 2');
+	`
+
+	h := harness.NewHarness(t, sql, "add-no-notification-alert-log")
+	defer h.Close()
+
+	doQL := func(t *testing.T, h *harness.Harness, query string, res interface{}) {
+		t.Helper()
+		g := h.GraphQLQueryUserT(t, h.UUID("user"), query)
+		for _, err := range g.Errors {
+			t.Error("GraphQL Error:", err.Message)
+		}
+		if len(g.Errors) > 0 {
+			t.Fatal("errors returned from GraphQL")
+		}
+		t.Log("Response:", string(g.Data))
+		if res == nil {
+			return
+		}
+		err := json.Unmarshal(g.Data, &res)
+		if err != nil {
+			t.Fatal("failed to parse response:", err)
+		}
+	}
+
+	// favorite the second service from initSQL
+	doQL(t, h, fmt.Sprintf(`
+		mutation{
+			setFavorite(input:{
+				target:{
+					id: "%s"
+					type: service
+				}
+				favorite: true
+			})
+		}
+	`, h.UUID("sid2")), nil)
+
+	var s struct {
+		Service struct {
+			IsFavorite bool
+		}
+	}
+
+	// assert the second service was favorited
+	doQL(t, h, fmt.Sprintf(`
+		query {
+			service(id: "%s") { 
+				isFavorite 
+			}	
+		}
+	`, h.UUID("sid2")), &s)
+
+	if s.Service.IsFavorite != true {
+		t.Fatalf("ERROR: ServiceID %s IsUserFavorite=%t; want true", h.UUID("sid2"), s.Service.IsFavorite)
+	}
+
+	// create alerts against both services (notifed version & favorited version)
+	h.CreateAlert(h.UUID("sid"), "notified alert")
+	h.CreateAlert(h.UUID("sid2"), "favorited alert")
+
+	type Alerts struct {
+		Alerts struct {
+			Nodes []struct {
+				ID string
+			}
+		}
+	}
+
+	var alerts1, alerts2, alerts3 Alerts
+
+	// output: 1 alert (the favorited one)
+	doQL(t, h, `query {
+		alerts(input: {
+			includeNotified: false
+			favoritesOnly: true
+		}) {
+			nodes {
+				id
+				summary
+			}
+		}
+	}`, &alerts1)
+
+	if len(alerts1.Alerts.Nodes) != 1 {
+		t.Errorf("got %d alerts; want 1", len(alerts1.Alerts.Nodes))
+	}
+
+	// Expect 1 SMS for the created alert
+	h.Twilio(t).Device(h.Phone("1")).ExpectSMS("notified")
+
+	// query for favorites & notified: 2 alerts
+	doQL(t, h, `query {
+			alerts(input: {
+				includeNotified: true
+				favoritesOnly: true
+			}) {
+				nodes {
+					id
+					summary
+				}
+			}
+		}`, &alerts2)
+
+	if len(alerts2.Alerts.Nodes) != 2 {
+		t.Errorf("got %d alerts; want 2", len(alerts2.Alerts.Nodes))
+	}
+
+	// All Services test (favoritesOnly: false)
+	// output: 2 alerts
+	doQL(t, h, `query {
+		alerts(input: {
+			includeNotified: true
+			favoritesOnly: false
+		}) {
+			nodes {
+				id
+				summary
+			}
+		}
+	}`, &alerts3)
+
+	if len(alerts3.Alerts.Nodes) != 2 {
+		t.Errorf("got %d alerts; want 2", len(alerts3.Alerts.Nodes))
+	}
```

Comment:
Same

Reply:
Replaced with the assertion library version, and recommitted.